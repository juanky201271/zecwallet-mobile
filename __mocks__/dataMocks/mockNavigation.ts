import { StackScreenProps } from '@react-navigation/stack';

export const mockNavigation: StackScreenProps<any>['navigation'] = {
  // Propiedades y métodos necesarios para la navegación
  navigate: jest.fn(),
  goBack: jest.fn(),
  dispatch: jest.fn(),
  reset: jest.fn(),
  isFocused: jest.fn(),
  canGoBack: jest.fn(),
  getParent: jest.fn(),
  getId: jest.fn(),
  getState: jest.fn(),
  setParams: jest.fn(),
  setOptions: jest.fn(),
  addListener: jest.fn(),
  removeListener: jest.fn(),
  replace: jest.fn(),
  push: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
  // Agrega cualquier otra propiedad o método necesario para tu caso
};
