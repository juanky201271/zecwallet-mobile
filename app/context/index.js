import { defaultAppContextLoaded, ContextAppLoaded, ContextAppLoadedProvider } from './contextAppLoaded';
import { defaultAppContextLoading, ContextAppLoading, ContextAppLoadingProvider } from './contextAppLoading';

export {
  defaultAppContextLoaded,
  ContextAppLoaded,
  ContextAppLoadedProvider,
  defaultAppContextLoading,
  ContextAppLoading,
  ContextAppLoadingProvider,
};
