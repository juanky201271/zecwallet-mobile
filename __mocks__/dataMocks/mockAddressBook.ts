import { AddressBookFileClass } from '../../app/AppState';

export const mockAddressBook: AddressBookFileClass[] = [
  {
    label: 'pepe',
    address: 'u1234567890_____________',
  },
  {
    label: 'lolo',
    address: 'u0987654321_____________',
  },
];
