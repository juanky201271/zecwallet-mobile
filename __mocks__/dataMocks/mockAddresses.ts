import { AddressClass, AddressKindEnum, ReceiverEnum } from '../../app/AppState';

export const mockAddresses: AddressClass[] = [
  {
    uaAddress: 'UA-12345678901234567890',
    address: 'UA-12345678901234567890',
    addressKind: AddressKindEnum.u,
    containsPending: false,
    receivers: ReceiverEnum.o + ReceiverEnum.z + ReceiverEnum.t,
  },
  {
    uaAddress: 'UA-12345678901234567890',
    address: 'sapling-12345678901234567890',
    addressKind: AddressKindEnum.z,
    containsPending: false,
    receivers: ReceiverEnum.z,
  },
  {
    uaAddress: 'UA-12345678901234567890',
    address: 'transparent-12345678901234567890',
    addressKind: AddressKindEnum.t,
    containsPending: false,
    receivers: ReceiverEnum.t,
  },
];
