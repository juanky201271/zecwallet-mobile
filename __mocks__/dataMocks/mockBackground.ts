import { BackgroundType } from '../../app/AppState';

export const mockBackground: BackgroundType = {
  batches: 0,
  message: '',
  date: 0,
  dateEnd: 0,
};
