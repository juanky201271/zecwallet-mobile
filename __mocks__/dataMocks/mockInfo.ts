import { ChainNameEnum, CurrencyNameEnum, InfoType } from '../../app/AppState';
import { serverUris } from '../../app/uris';

export const mockInfo: InfoType = {
  chainName: ChainNameEnum.mainChainName,
  serverUri: serverUris(() => {})[0].uri,
  latestBlock: 2000000,
  connections: 0,
  version: 'server_version',
  verificationProgress: 0,
  currencyName: CurrencyNameEnum.ZEC,
  solps: 0,
  zingolib: 'zingolib_version',
};
