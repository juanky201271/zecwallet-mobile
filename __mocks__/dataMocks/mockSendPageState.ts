import { SendPageStateClass, ToAddrClass } from '../../app/AppState';
import Utils from '../../app/utils';

const mockSendPageState: SendPageStateClass = new SendPageStateClass(new ToAddrClass(Utils.getNextToAddrID()));

mockSendPageState.toaddr.id = 1234567890;
mockSendPageState.toaddr.to = 'UA-12345678901234567890';
mockSendPageState.toaddr.amount = '1.12345678';
mockSendPageState.toaddr.amountCurrency = '50.22';
mockSendPageState.toaddr.memo = 'memo1\nmemo2\nmemo3\nmemo4\nmemo5\nmemo6\nmemo7\nmemo8\nmemo9\nmemo10';

export default mockSendPageState;
