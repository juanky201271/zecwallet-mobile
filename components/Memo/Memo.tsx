/* eslint-disable react-native/no-inline-styles */
import React, { useCallback, useContext, useState } from 'react';
import {
  View,
  ScrollView,
  SafeAreaView,
  TextInput,
  Dimensions,
  TouchableOpacity,
  KeyboardAvoidingView,
  Platform,
} from 'react-native';
import { useTheme } from '@react-navigation/native';
import Button from '../Components/Button';
import { ThemeType } from '../../app/types';
import { ContextAppLoaded } from '../../app/context';
import Header from '../Header';
import moment from 'moment';
import 'moment/locale/es';
import 'moment/locale/pt';
import 'moment/locale/ru';
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
import { faXmark } from '@fortawesome/free-solid-svg-icons';
import { ButtonTypeEnum, GlobalConst } from '../../app/AppState';
import FadeText from '../Components/FadeText';
import { Buffer } from 'buffer';

type MemoProps = {
  closeModal: () => void;
  memoUpdateToField: (memo: string | null) => void;
};
const Memo: React.FunctionComponent<MemoProps> = ({ closeModal, memoUpdateToField }) => {
  const context = useContext(ContextAppLoaded);
  const { translate, sendPageState, language, uaAddress } = context;
  const { colors } = useTheme() as unknown as ThemeType;
  moment.locale(language);

  const [memo, setMemo] = useState<string>(sendPageState.toaddr.memo);

  const includeUAMemo = sendPageState.toaddr.includeUAMemo;

  const dimensions = {
    width: Dimensions.get('window').width,
    height: Dimensions.get('window').height,
  };

  const doSaveAndClose = () => {
    memoUpdateToField(memo);
    closeModal();
  };

  const memoTotal = useCallback(
    (memoStr: string, includeUAMemoBoo: boolean) => {
      return `${memoStr || ''}${includeUAMemoBoo ? '\nReply to: \n' + uaAddress : ''}`;
    },
    [uaAddress],
  );

  const countMemoBytes = (memoStr: string, includeUAMemoBoo: boolean) => {
    const len = Buffer.byteLength(memoTotal(memoStr, includeUAMemoBoo), 'utf8');
    return len;
  };

  return (
    <KeyboardAvoidingView
      behavior={Platform.OS === GlobalConst.platformOSios ? 'padding' : 'height'}
      keyboardVerticalOffset={Platform.OS === GlobalConst.platformOSios ? 10 : 0}
      style={{ backgroundColor: colors.background }}>
      <SafeAreaView
        style={{
          display: 'flex',
          justifyContent: 'flex-start',
          alignItems: 'stretch',
          height: '100%',
          backgroundColor: colors.background,
        }}>
        <Header
          title={translate('send.memo') as string}
          noBalance={true}
          noSyncingStatus={true}
          noDrawMenu={true}
          noPrivacy={true}
          closeScreen={closeModal}
        />
        <ScrollView
          style={{
            height: '80%',
            maxHeight: '80%',
            minHeight: '50%',
          }}
          contentContainerStyle={{
            flexDirection: 'column',
            alignItems: 'stretch',
            justifyContent: 'flex-start',
            padding: 20,
          }}>
          <View
            accessible={true}
            accessibilityLabel={translate('send.memo-acc') as string}
            style={{
              flexGrow: 1,
              borderWidth: 1,
              borderRadius: 5,
              borderColor: colors.text,
              minWidth: 48,
              minHeight: 48,
              maxHeight: dimensions.height * 0.4,
              flexDirection: 'row',
            }}>
            <TextInput
              testID="send.memo-field"
              multiline
              style={{
                flex: 1,
                color: colors.text,
                fontWeight: '600',
                fontSize: 14,
                minWidth: 48,
                minHeight: 48,
                margin: 5,
                backgroundColor: 'transparent',
                textAlignVertical: 'top',
              }}
              value={memo}
              onChangeText={(text: string) => setMemo(text)}
              editable={true}
              maxLength={GlobalConst.memoMaxLength}
            />
            {memo && (
              <TouchableOpacity
                onPress={() => {
                  setMemo('');
                }}>
                <FontAwesomeIcon style={{ margin: 10 }} size={25} icon={faXmark} color={colors.primaryDisabled} />
              </TouchableOpacity>
            )}
          </View>
          <View
            style={{
              flexDirection: 'row',
              justifyContent: 'flex-end',
              alignItems: 'center',
            }}>
            <FadeText
              style={{
                marginTop: 0,
                fontWeight: 'bold',
                color: countMemoBytes(memo, includeUAMemo) > GlobalConst.memoMaxLength ? 'red' : colors.text,
              }}>{`${countMemoBytes(memo, includeUAMemo)} `}</FadeText>
            <FadeText style={{ marginTop: 0 }}>{translate('loadedapp.of') as string}</FadeText>
            <FadeText style={{ marginTop: 0 }}>{' ' + GlobalConst.memoMaxLength.toString() + ' '}</FadeText>
          </View>
        </ScrollView>
        <View
          style={{
            flexGrow: 1,
            flexDirection: 'row',
            justifyContent: 'center',
            alignItems: 'center',
            marginVertical: 10,
          }}>
          <Button
            type={ButtonTypeEnum.Primary}
            title={translate('save') as string}
            onPress={doSaveAndClose}
            disabled={countMemoBytes(memo, includeUAMemo) > GlobalConst.memoMaxLength}
          />
        </View>
      </SafeAreaView>
    </KeyboardAvoidingView>
  );
};

export default Memo;
