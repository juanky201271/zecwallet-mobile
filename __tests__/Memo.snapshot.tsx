/**
 * @format
 */

import 'react-native';
import React from 'react';

import { render } from '@testing-library/react-native';
import Memo from '../components/Memo';
import { defaultAppContextLoaded, ContextAppLoadedProvider } from '../app/context';
import { mockTranslate } from '../__mocks__/dataMocks/mockTranslate';
import mockSendPageState from '../__mocks__/dataMocks/mockSendPageState';

jest.useFakeTimers();
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));
jest.mock('react-native-localize', () => ({
  getNumberFormatSettings: () => {
    return {
      decimalSeparator: '.',
      groupingSeparator: ',',
    };
  },
}));
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.mock('@react-native-community/netinfo', () => {
  const RN = jest.requireActual('react-native');

  RN.NativeModules.RNCNetInfo = {
    execute: jest.fn(() => '{}'),
  };

  return RN;
});
jest.mock('react-native', () => {
  const RN = jest.requireActual('react-native');

  RN.NativeModules.RPCModule = {
    execute: jest.fn(() => '{}'),
  };

  return RN;
});

// test suite
describe('Component Memo - test', () => {
  //snapshot test
  test('Memo - snapshot', () => {
    const state = defaultAppContextLoaded;
    state.translate = mockTranslate;
    state.sendPageState = mockSendPageState;
    const onClose = jest.fn();
    const memo = render(
      <ContextAppLoadedProvider value={state}>
        <Memo closeModal={onClose} updateToField={onClose} />
      </ContextAppLoadedProvider>,
    );
    expect(memo.toJSON()).toMatchSnapshot();
  });
});
