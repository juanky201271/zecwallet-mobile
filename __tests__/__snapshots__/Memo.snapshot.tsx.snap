// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Memo - test Memo - snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "stretch",
      "backgroundColor": "rgb(242, 242, 242)",
      "display": "flex",
      "height": "100%",
      "justifyContent": "flex-start",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "alignItems": "stretch",
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "padding": 20,
      }
    }
    style={
      {
        "maxHeight": "70%",
        "minHeight": "50%",
      }
    }
  >
    <View>
      <View
        accessibilityLabel="text translated"
        accessible={true}
        style={
          {
            "borderColor": "rgb(28, 28, 30)",
            "borderRadius": 5,
            "borderWidth": 1,
            "flexDirection": "row",
            "flexGrow": 1,
            "maxHeight": 533.6,
            "minHeight": 400.2,
            "minWidth": 48,
          }
        }
      >
        <TextInput
          editable={true}
          maxLength={511}
          multiline={true}
          onChangeText={[Function]}
          style={
            {
              "backgroundColor": "transparent",
              "color": "rgb(28, 28, 30)",
              "flex": 1,
              "fontSize": 14,
              "fontWeight": "600",
              "margin": 5,
              "minHeight": 48,
              "minWidth": 48,
              "textAlignVertical": "top",
            }
          }
          testID="send.memo-field"
          value="memo1
memo2
memo3
memo4
memo5
memo6
memo7
memo8
memo9
memo10"
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <
            icon={
              {
                "icon": [
                  384,
                  512,
                  [
                    128473,
                    10005,
                    10006,
                    10060,
                    215,
                    "close",
                    "multiply",
                    "remove",
                    "times",
                  ],
                  "f00d",
                  "M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z",
                ],
                "iconName": "xmark",
                "prefix": "fas",
              }
            }
            size={25}
            style={
              {
                "margin": 10,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <Text
          style={
            {
              "color": "rgb(28, 28, 30)",
              "fontWeight": "bold",
              "marginTop": 5,
              "opacity": 0.65,
            }
          }
        >
          60 
        </Text>
        <Text
          style={
            {
              "color": "rgb(28, 28, 30)",
              "marginTop": 5,
              "opacity": 0.65,
            }
          }
        >
          text translated
        </Text>
        <Text
          style={
            {
              "color": "rgb(28, 28, 30)",
              "marginTop": 5,
              "opacity": 0.65,
            }
          }
        >
           511 
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 1,
        "justifyContent": "center",
        "marginVertical": 5,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgb(0, 122, 255)",
          "borderColor": "rgb(28, 28, 30)",
          "borderRadius": 10,
          "borderWidth": 2,
          "justifyContent": "center",
          "maxWidth": "90%",
          "minHeight": 48,
          "minWidth": "30%",
          "opacity": 1,
          "padding": 0,
          "paddingLeft": 20,
          "paddingRight": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 0,
            "minHeight": 48,
            "minWidth": 48,
            "padding": 0,
          }
        }
      >
        <Text
          style={
            {
              "color": "rgb(242, 242, 242)",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "center",
              "textTransform": "uppercase",
            }
          }
        >
          text translated
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgb(242, 242, 242)",
          "borderColor": "rgb(0, 122, 255)",
          "borderRadius": 10,
          "borderWidth": 2,
          "justifyContent": "center",
          "marginLeft": 10,
          "maxWidth": "90%",
          "minHeight": 48,
          "minWidth": "30%",
          "opacity": 1,
          "padding": 0,
          "paddingLeft": 20,
          "paddingRight": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 0,
            "minHeight": 48,
            "minWidth": 48,
            "padding": 0,
          }
        }
      >
        <Text
          style={
            {
              "color": "rgb(0, 122, 255)",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "center",
              "textTransform": "uppercase",
            }
          }
        >
          text translated
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
